commit cacea62fcdda5656cb5b8104e73a00e043b61730
Author: Vladimir Oltean <vladimir.oltean@nxp.com>
Date:   Fri Jan 29 03:00:03 2021 +0200

    net: mscc: ocelot: don't use NPI tag prefix for the CPU port module
    
    Context: Ocelot switches put the injection/extraction frame header in
    front of the Ethernet header. When used in NPI mode, a DSA master would
    see junk instead of the destination MAC address, and it would most
    likely drop the packets. So the Ocelot frame header can have an optional
    prefix, which is just "ff:ff:ff:ff:ff:fe > ff:ff:ff:ff:ff:ff" padding
    put before the actual tag (still before the real Ethernet header) such
    that the DSA master thinks it's looking at a broadcast frame with a
    strange EtherType.
    
    Unfortunately, a lesson learned in commit 69df578c5f4b ("net: mscc:
    ocelot: eliminate confusion between CPU and NPI port") seems to have
    been forgotten in the meanwhile.
    
    The CPU port module and the NPI port have independent settings for the
    length of the tag prefix. However, the driver is using the same variable
    to program both of them.
    
    There is no reason really to use any tag prefix with the CPU port
    module, since that is not connected to any Ethernet port. So this patch
    makes the inj_prefix and xtr_prefix variables apply only to the NPI
    port (which the switchdev ocelot_vsc7514 driver does not use).
    
    Signed-off-by: Vladimir Oltean <vladimir.oltean@nxp.com>
    Reviewed-by: Florian Fainelli <f.fainelli@gmail.com>
    Signed-off-by: Jakub Kicinski <kuba@kernel.org>

diff --git a/include/soc/mscc/ocelot.h b/include/soc/mscc/ocelot.h
index cdc33fa05660..93c22627dedd 100644
--- a/include/soc/mscc/ocelot.h
+++ b/include/soc/mscc/ocelot.h
@@ -651,8 +651,8 @@ struct ocelot {
 
 	int				npi;
 
-	enum ocelot_tag_prefix		inj_prefix;
-	enum ocelot_tag_prefix		xtr_prefix;
+	enum ocelot_tag_prefix		npi_inj_prefix;
+	enum ocelot_tag_prefix		npi_xtr_prefix;
 
 	u32				*lags;